### PR TITLE
Ltd 2150 initial refactoring

### PR DESCRIPTION
**Pull request by kevincarrogan**

### CODE REVIEW

File: exporter/applications/views/goods/add_good_firearm/views/edit.py [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+
+
+class BaseEditView(LoginRequiredMixin, FormView):
+    template_name = "core/form.html"
+
+    @cached_property
+    def application_id(self):
+        return str(self.kwargs["pk"])
+
+    @cached_property
```
Comment (depsiatwal) [inline exporter/applications/views/goods/add_good_firearm/views/edit.py:72]: is it worth having this Cached properties as a mixin as I notice it's used 4 times in this file ?.

File: exporter/applications/views/goods/add_good_firearm/views/edit.py [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+                    status_code,
+                    api_resp_data,
+                    "Error updating the product document description - response was: %s - %s",
+                    "Unexpected error updating product document description",
+                )
+
+        return redirect(self.get_success_url())
+
+
+class AddGoodWizardCommon:
```
Comment (depsiatwal) [inline exporter/applications/views/goods/add_good_firearm/views/edit.py:322]: is this better named as mixin to make it Django like  ?

File: exporter/applications/views/goods/add_good_firearm/views/edit.py [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+        if status_code != HTTPStatus.OK:
+            raise ServiceError(
+                status_code,
+                api_resp_data,
+                "Error updating firearm - response was: %s - %s",
+                "Unexpected error updating firearm",
+            )
+
+
+class AddGoodDocumentWizardCommon(AddGoodWizardCommon):
```
Comment (depsiatwal) [inline exporter/applications/views/goods/add_good_firearm/views/edit.py:390]: two commons ? 
personally I find it confusing but may be it's just me

**Review by depsiatwal — COMMENTED** [review]
few commons also does it make more sense to move all the extra classes that don't contain views in the /core/ subfolder ?

File: exporter/applications/views/goods/add_good_firearm/views/edit.py (hunk 1 above)
Comment (kevincarrogan) [reply exporter/applications/views/goods/add_good_firearm/views/edit.py:72]: Yes. We will eventually do that, Arun has already done some of that but I think we are waiting until these views have stabilised.

**Comment by kevincarrogan** [comment]
No. I don’t think it makes sense to move them to `core`. This code is only relevant (right now at least) to these views. Moving it to `core` moves the code further away from its place of use and infers it’s more common than it really is.

**Review by depsiatwal — APPROVED** [review]
LGTM

File: exporter/applications/views/goods/add_good_firearm/views/edit.py (hunk 3 above)
Comment (saruniitr) [reply exporter/applications/views/goods/add_good_firearm/views/edit.py:390]: these can be renamed with better names, currently these views are not fully stable.

**Pull request merged**